File: examples/python/09-new-features/barcode_svg/main.py
# Barcode SVG generation
#
# Demonstrates generating 1D barcodes and QR codes as vector SVG strings.
# Run: python main.py

from __future__ import annotations

import os

import pdf_oxide


OUT_DIR = "output"


def main() -> None:
    os.makedirs(OUT_DIR, exist_ok=True)

    # 1D barcode — Code 128 SVG (barcode_type=0)
    svg = pdf_oxide.generate_barcode_svg(0, "PDF-OXIDE-0341")
    assert svg.startswith("<svg"), f"expected SVG, got: {svg[:40]}"
    path = os.path.join(OUT_DIR, "code128.svg")
    with open(path, "w") as f:
        f.write(svg)
    print(f"Written: {path} ({len(svg)} bytes)")

    # 1D barcode — EAN-13 SVG (barcode_type=2)
    svg = pdf_oxide.generate_barcode_svg(2, "5901234123457")
    assert svg.startswith("<svg")
    path = os.path.join(OUT_DIR, "ean13.svg")
    with open(path, "w") as f:
        f.write(svg)
    print(f"Written: {path} ({len(svg)} bytes)")

    # Data Matrix SVG (size=256)
    svg = pdf_oxide.generate_datamatrix_svg("https://github.com/yfedoseev/pdf_oxide", 1, 256)
    assert svg.startswith("<svg")
    assert "<rect" in svg, "QR SVG must contain rect elements"
    path = os.path.join(OUT_DIR, "datamatrix_code.svg")
    with open(path, "w") as f:
        f.write(svg)
    print(f"Written: {path} ({len(svg)} bytes)")

    # QR code SVG (error_correction=1=Medium, size=256)
    svg = pdf_oxide.generate_qr_svg("https://github.com/yfedoseev/pdf_oxide", 1, 256)
    assert svg.startswith("<svg")
    assert "<rect" in svg, "QR SVG must contain rect elements"
    path = os.path.join(OUT_DIR, "qr_code.svg")
    with open(path, "w") as f:
        f.write(svg)
    print(f"Written: {path} ({len(svg)} bytes)")

    print("All barcode SVG checks passed.")


if __name__ == "__main__":
    main()

File: examples/rust/09-new-features/barcode_svg/main.rs
// Barcode SVG generation
//
// Demonstrates generating 1D barcodes and QR codes as scalable SVG strings
// (vector output, no rasterisation). Requires the `barcodes` feature.
//
// Run: cargo run --example showcase_barcode_svg --features barcodes

use pdf_oxide::{
    error::Result,
    writer::{BarcodeGenerator, BarcodeOptions, BarcodeType, DataMatrixOptions, QrCodeOptions},
};
use std::path::PathBuf;

fn main() -> Result<()> {
    let out_dir = PathBuf::from("target/examples_output/barcode_svg");
    std::fs::create_dir_all(&out_dir)?;

    // 1D barcode — Code 128 SVG
    let svg = BarcodeGenerator::generate_1d_svg(
        BarcodeType::Code128,
        "PDF-OXIDE-0341",
        &BarcodeOptions::default().width(400).height(80),
    )?;
    assert!(svg.starts_with("<svg"), "expected SVG output for Code128");
    let path = out_dir.join("code128.svg");
    std::fs::write(&path, &svg)?;
    println!("Written: {} ({} bytes)", path.display(), svg.len());

    // 1D barcode — EAN-13 SVG
    let svg = BarcodeGenerator::generate_1d_svg(
        BarcodeType::Ean13,
        "5901234123457",
        &BarcodeOptions::default().width(300).height(80),
    )?;
    assert!(svg.starts_with("<svg"));
    let path = out_dir.join("ean13.svg");
    std::fs::write(&path, &svg)?;
    println!("Written: {} ({} bytes)", path.display(), svg.len());

    // Data Matrix SVG
    let svg = BarcodeGenerator::generate_datamatrix_svg(
        "https://github.com/yfedoseev/pdf_oxide",
        &DataMatrixOptions::default().size(256),
    )?;
    assert!(svg.starts_with("<svg"));
    assert!(svg.contains("<rect"), "Data Matrix SVG must contain rect elements");
    let path = out_dir.join("datamatrix.svg");
    std::fs::write(&path, &svg)?;
    println!("Written: {} ({} bytes)", path.display(), svg.len());

    // QR code SVG
    let svg = BarcodeGenerator::generate_qr_svg(
        "https://github.com/yfedoseev/pdf_oxide",
        &QrCodeOptions::default().size(256),
    )?;
    assert!(svg.starts_with("<svg"));
    assert!(svg.contains("<rect"), "QR SVG must contain rect elements");
    let path = out_dir.join("qr_code.svg");
    std::fs::write(&path, &svg)?;
    println!("Written: {} ({} bytes)", path.display(), svg.len());

    println!("All barcode SVG checks passed.");
    Ok(())
}
